### PR TITLE
E2E matchup

### PR DESCRIPTION
**Pull request by holisticode**
avalanchego relies on avalanche-testing for running the end-to-end tests. The matchup of avalanchego:version with avalanche-testing:version is done manually. This brings undesirable scenarios where matchups are incorrect, or updates to avalanchego are not replicated to avalanche-testing, etc.
The problem is defined as the lack of coupling between avalanchego and avalanche-testing.

The solution is defined as having a matchup with a ruleset between versions in avalachego and avalanche-testing.

Matching PR in avalanche-testing: https://github.com/ava-labs/avalanche-testing/pull/36

### CODE REVIEW

**Review by otherview — DISMISSED** [review]
LGTM

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ env:
   - E2E_TEST_HOME=$GOPATH/src/github.com/ava-labs/avalanche-testing/
   - COMMIT=${TRAVIS_COMMIT::8}
   - DOCKERHUB_REPO=avaplatform/avalanchego
+  - BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
```
Comment (otherview) [inline .travis.yml:13]: @holisticode @aaronbuchwald 
Can you review this ?

A lot of people are complaining about in https://github.com/travis-ci/travis-ci/issues/6652 ?

Comment (holisticode) [reply .travis.yml:13]: This looks like the solution suggested in https://unhashable.com/getting-the-current-branch-name-during-a-pull-request-in-travis-ci/. So if the desired branch for a PR is indeed

> the name of the branch from which the PR originated, if the current job is a pull request

then this looks good to me.

However, I doubt this works in a yml file like that and probably requires a shell script.

Comment (otherview) [reply .travis.yml:13]: Yeah makes sense all right - let me revert this 👍

**Pull request closed without merging**